cmd/agent/subcommands/run/command.go

```diff
@@ -267,6 +270,24 @@ func getSharedFxOption() fx.Option {
 			params.Options.EnabledFeatures = pkgforwarder.SetFeature(params.Options.EnabledFeatures, pkgforwarder.CoreFeatures)
 			return params
 		}),
+
+		// workloadmeta setup
+		collectors.GetCatalog(),
+		fx.Provide(func(config config.Component) workloadmeta.Params {
```

Comment:
This snippet is copy-pasted five times in this PR. Would it make sense to refactor into a helper function somewhere? (maybe in the component package?)

Reply:
I will address this in a follow-up PR.